src/runtime-helpers/runtime-helpers.ts

```diff
@@ -213,6 +227,37 @@ module ScriptLab {
     function _isDialogApiSupported() {
         return (window as any).Office.context.requirements.isSetSupported('DialogApi');
     }
+
+    function _extractAndCacheAccessToken(message: string, clientId: string, resource: string): string {
+        const authResponseKeyValues = message.split('&');
+        const accessToken = authResponseKeyValues[0].substr('AUTH:access_token='.length);
+        const expiresIn = Number(authResponseKeyValues[1].substr('AUTH:expires_in='.length));
+        cachedAuthTokens[clientId + resource] = { token: accessToken, expiry: Date.now() + ((expiresIn - 60) * 1000) };
+        return accessToken;
+    }
+
+    function _getCachedAccessToken(clientId: string, resource: string): string {
+        const cachedAuthToken = cachedAuthTokens[clientId + resource];
+        if (cachedAuthToken) {
```

Comment:
I prefer functions that bail out fast.  So instead of phrasing the `if` as a positive and then having an `else` that is a one-liner `return null`, I would reverse it to be:

    if (!cachedAuthToken) {
        return null;
    }

And then de-nest the rest of the logic.